src/main/java/io/pravega/connectors/flink/sink/PravegaTransactionalWriter.java

```diff
@@ -159,6 +164,7 @@ public void write(T element, Context context) throws IOException, InterruptedExc
         } catch (TxnFailedException | AssertionError e) {
             throw new IOException(e);
         }
+        numRecordsOutCounter.inc();
```

Comment:
If a transaction is aborted, this would have accounted the records of the failed transactions as successful records written. Is that the intended goal here?

Reply:
In the sink API a transaction is only aborted when we close the writer. When the transaction fails, it should throw an exception here so the counter won't increase.